types/index.d.ts

```diff
@@ -0,0 +1,87 @@
+// Minimum TypeScript Version: 3.2
+import {Plugin} from 'unified'
+
+/**
+ * Wrap a document around a fragment.
+ */
+declare const document: Plugin<[document.Options?]>
+
+declare namespace document {
+  interface Options {
+    /**
+     * Text to use as title
+     *
+     * default: name of file, if any
+     */
+    title?: string
+
+    /**
+     * Natural language of document.
+     *
+     * should be a [BCP 47](https://tools.ietf.org/html/bcp47) language tag.
+     *
+     * @default 'en'
+     */
+    language?: string
+
+    /**
+     * [Doctype](https://github.com/wooorm/doctype) to use.
+     *
+     * @default '5'
+     */
+    doctype?: string
+
+    /**
+     * CSS to include in `head` in `<style>` elements.
+     *
+     * @default []
+     */
+    style?: string | string[]
+
+    /**
+     * Links to stylesheets to include in `head`
+     *
+     * @default []
+     */
+    css?: string | string[]
+
+    /**
+     * Metadata to include in `head`.
+     *
+     * Each object is passed as
+     * [`properties`](https://github.com/syntax-tree/hastscript#hselector-properties-children)
+     * to [`hastscript`](https://github.com/syntax-tree/hastscript) with a
+     * `meta` element.
+     *
+     * @default []
+     */
+    meta?: Record<string, any> | Array<Record<string, any>>
+
+    /**
+     * Link tags to include in `head`.
+     *
+     * Each object is passed as
+     * [`properties`](https://github.com/syntax-tree/hastscript#hselector-properties-children)
+     * to [`hastscript`](https://github.com/syntax-tree/hastscript) with a `link` element.
+     *
+     * @default []
+     */
+    link?: Record<string, any> | Array<Record<string, any>>
```

Comment:
Based of the tests `link` also appears to be string keys and values
https://github.com/rehypejs/rehype-document/blob/498a1669121f9cbd3241f550712ab5654956fec6/test.js#L235-L245
```suggestion
    link?: Record<string, string> | Array<Record<string, string>>
```